### PR TITLE
edit an API

### PR DESCRIPTION
**Pull request by johnayinde**
- added **logo_url, api_website and term_of_use**  columns  to entity file
- 
- create DTO file showing all the optional columns that can be updated on the API Table
- Generated migration file for the updated entity

### CODE REVIEW

**Comment by Chimexx** [comment]
image
This is what I get, if it works fine on your end, I'll like to see a screenshot

**Comment by johnayinde** [comment]
> image This is what I get, if it works fine on your end, I'll like to see a screenshot

**Comment by johnayinde** [comment]
@Chimexx 
apiNAme

category

you probably used a wrongly  formatted UUID string
 try and change the first or any number/character in a valid ID to a wrong one

File: src/api/api.service.ts [hunk 1]
```diff
@@ -32,89 +37,162 @@ export class ApiService {
...
+    profileId: string,
+    updateApiDto: UpdateApiDto,
+  ): Promise<Api> {
+    try {
+      const api = await this.apiRepo.findOne({ where: { id: apiId } });
+      if (api) {
+        /* Checking if the user is the owner of the api. */
+        const verified = await this.verify(apiId, profileId);
+        if (verified === false) {
+          throw new NotFoundException(
```
Comment (Chimexx) [inline src/api/api.service.ts:142]: You haven't fixed this, it should be BadRequestException

File: src/api/api.service.ts [hunk 2]
```diff
@@ -32,89 +37,162 @@ export class ApiService {
...
-                ZaLaResponse.BadRequest('Internal Server error', err.message, '500'),
-            );
+
+        /* Checking if categoryID exist in the api. */
+        const findCategory = await this.apiRepo.findOne({
+          where: { categoryId: updateApiDto.categoryId },
+        });
+
+        if (!findCategory) {
+          throw new NotFoundException(
```
Comment (Chimexx) [inline src/api/api.service.ts:153]: What will be the effect of this block of code assuming the user isn't updating the category? This should only run if the user is updating category too

File: src/api/api.service.ts [hunk 3]
```diff
@@ -32,89 +37,162 @@ export class ApiService {
...
-
-            if(apiExist && isOwner === true) {
-                return await this.apiRepo.remove(apiExist);
-            }
-            throw new NotFoundException(
-                ZaLaResponse.NotFoundRequest('Not Found', 'Api does not exist', '404'),
-            ); 
+        /* Checking if the new category name already exist. */
+        const apiNameExist = await this.apiRepo.findOne({
+          where: { name: updateApiDto.name },
```
Comment (Chimexx) [inline src/api/api.service.ts:164]: You don't have to check category name, what you are updating is api not category

File: src/api/api.service.ts [hunk 4]
```diff
@@ -32,89 +37,162 @@ export class ApiService {
...
+        });
+
+        if (apiNameExist) {
+          throw new BadRequestException(
+            ZaLaResponse.BadRequest(
```
Comment (Chimexx) [inline src/api/api.service.ts:169]: This block of code forces the user to update the category whether he likes it or not. rremove it.

**Review by Chimexx — CHANGES_REQUESTED** [review]

File: src/api/api.service.ts [hunk 5]
```diff
@@ -32,89 +37,162 @@ export class ApiService {
...
+        });
+
+        if (!findCategory) {
+          throw new NotFoundException(
+            ZaLaResponse.NotFoundRequest(
+              'Not Found',
+              'Category does not exist',
+              '404',
+            ),
+          );
```
Comment (Chimexx) [inline src/api/api.service.ts:159]: What will be the effect of this block of code assuming the user isn't updating the category? This should only run if the user is updating category too

File: src/api/api.service.ts [hunk 6]
```diff
@@ -32,89 +37,162 @@ export class ApiService {
...
+          where: { name: updateApiDto.name },
+        });
+
+        if (apiNameExist) {
+          throw new BadRequestException(
+            ZaLaResponse.BadRequest(
+              'Existing values',
+              'An api with this name already exist... try another name',
+            ),
+          );
```
Comment (Chimexx) [inline src/api/api.service.ts:173]: This block of code forces the user to change the api name whether he likes it or not. rremove it.

File: src/api/api.service.ts [hunk 7]
```diff
@@ -147,18 +147,26 @@ export class ApiService {
           );
         }
 
-        /* Checking if the new updated API name already exist. */
-        const apiNameExist = await this.apiRepo.findOne({
-          where: { name: updateApiDto.name },
-        });
+        console.log(updateApiDto.name);
```
Comment (SunmadeBabafemi) [inline src/api/api.service.ts:150]: anytime you use console.logs for debugging and the likes try to clean it up before pushing it...

**Review by SunmadeBabafemi — CHANGES_REQUESTED** [review]

**Review by PhilemonBrain — DISMISSED** [review]

**Review by PhilemonBrain — APPROVED** [review]

**Pull request merged**